### PR TITLE
Refunds Step 2 - Networking layer, Mark I

### PR DESCRIPTION
**Pull request by mindgraffiti**
Partial work for #1307.

This PR adds the condensed version of refunds to an Order model in Networking. Unit testing via Circle CI will fail. Only local testing on Xcode will pass. I'm stopping here because this PR is already too big.

- removed previously written Networking code 
- added real JSON response for a fully refunded order
- added real JSON response for a partially refunded order
- added OrderRefundCondensed model to Networking and to Order model
- added OrderMapper unit tests
- added a Refunds feature flag

### To test
1. Target the Networking subproject
2. Run all unit tests for subproject

All Networking unit tests should pass.

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/System/FeatureFlag.swift [hunk 1]
```diff
@@ -26,6 +30,8 @@ enum FeatureFlag: Int {
             return BuildConfiguration.current == .localDeveloper
         case .reviews:
             return false
+        case .reviews:
```
Comment (jaclync) [inline WooCommerce/Classes/System/FeatureFlag.swift]: `refunds`?

File: Networking/NetworkingTests/Mapper/OrderMapperTests.swift [hunk 2]
```diff
@@ -120,6 +120,90 @@ class OrderMapperTests: XCTestCase {
         let orderModifiedString = format.string(from: brokenOrder.dateModified)
         XCTAssertEqual(orderModifiedString, todayCreatedString)
     }
+
+    /// Verfies that the coupon fields for an Order are correctly parsed.
```
Comment (jaclync) [inline Networking/NetworkingTests/Mapper/OrderMapperTests.swift]: nit: Verifies

File: Networking/NetworkingTests/Mapper/OrderMapperTests.swift [hunk 3]
```diff
@@ -120,6 +120,90 @@ class OrderMapperTests: XCTestCase {
...
+    func testOrderRefundCondensedFieldsDoNotExistAreParsedCorrectly() {
+        guard let order = mapLoadOrderResponse() else {
+            XCTFail()
+            return
+        }
+
+        XCTAssertNil(order.refunds)
+    }
+
+    /// Verfies that an Order with refund fields are correctly parsed.
```
Comment (jaclync) [inline Networking/NetworkingTests/Mapper/OrderMapperTests.swift]: nit: Verifies

File: WooCommerce/Classes/System/FeatureFlag.swift (hunk 1 above)
Comment (mindgraffiti) [reply WooCommerce/Classes/System/FeatureFlag.swift]: 🤦‍♀ haha yes!

File: Networking/Networking/Model/Order.swift [hunk 4]
```diff
@@ -123,6 +126,9 @@ public struct Order: Decodable {
 
         let coupons = try container.decode([OrderCouponLine].self, forKey: .couponLines)
 
+        // The refunds field will not always exist in the response, so let's default to an emtpy array.
```
Comment (jaclync) [inline Networking/Networking/Model/Order.swift]: nit: emtpy -> empty :)

**Comment by mindgraffiti** [comment]
Override and merge. This PR will temporarily break the build and tests.

**Pull request merged**